File: src/tauri/commands/schedule.ts
import { safeInvoke } from "./_base";

export type WorkInfoEntry = {
  phase: string;
  job_id: string;
  job_name: string;
  hours: number;
  sheet_name: string;
};

export type DayWork = {
  day: string;
  entries: WorkInfoEntry[];
};

export type WorkerSchedule = {
  worker_name: string;
  days: DayWork[];
};

export type ScheduleResult = {
  file_path: string;
  target_month: string;
  workers: WorkerSchedule[];
};

export function readScheduleExcel(filePath: string, targetYear: number, targetMonth: number, userFilter?: string) {
  return safeInvoke<ScheduleResult>("read_schedule_excel", {
    filePath,
    targetYear,
    targetMonth,
    userFilter,
  });
}

File: src-tauri/src/models/schedule.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct WorkInfoEntry {
    pub phase: String,
    pub job_id: String,
    pub job_name: String,
    pub hours: f64,
    pub sheet_name: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DayWork {
    pub day: String,
    pub entries: Vec<WorkInfoEntry>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct WorkerSchedule {
    pub worker_name: String,
    pub days: Vec<DayWork>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ScheduleResult {
    pub file_path: String,
    pub target_month: String,
    pub workers: Vec<WorkerSchedule>,
}
